v1 epp

capri-xiyue opened the pull request:
This is a mirror of https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/1118

capri-xiyue committed:
changed epp and conformance test to support v1

capri-xiyue committed:
fixed duplicate import

capri-xiyue committed:
changed docker file

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
fixed integration test

capri-xiyue committed:
fixed e2e test

capri-xiyue committed:
change manifests

capri-xiyue committed:
added rbac

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
tried all alias

capri-xiyue committed:
push local changes

capri-xiyue committed:
changed inferencepoollist not to use v1.InferencePoolList

capri-xiyue committed:
changed comments

capri-xiyue committed:
fixed presubmit

capri-xiyue committed:
change to v1 ip

capri-xiyue committed:
revert to corev1 objectreference

capri-xiyue committed:
revert back to corev1 objectreference

capri-xiyue committed:
change crd

capri-xiyue committed:
remove crd

capri-xiyue committed:
manually add x-k8s.io inferencepool

capri-xiyue committed:
revert crd

capri-xiyue committed:
Revert "added rbac"

This reverts commit c01cbc8a2a96f6aad519749a1ef95f5f3fa33090.

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
run generator

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
revert to workable

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
fixed test

capri-xiyue committed:
test

capri-xiyue committed:
revert it back to originall workable version

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
changed unused code

capri-xiyue committed:
fixed pipeline:

capri-xiyue committed:
fixed missing dependecny update

capri-xiyue committed:
fixed imports

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
merge to v1-inferencepool

capri-xiyue committed:
fixed integration test

capri-xiyue committed:
fixed import

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue commented on conformance/resources/manifests/manifests.yaml:
```diff
@@ -290,7 +290,7 @@ spec:
       containers:
       - name: epp
-        image: us-central1-docker.pkg.dev/k8s-staging-images/gateway-api-inference-extension/epp:main
+        image: us-west1-docker.pkg.dev/xiyue-gke-dev-new/k8s-dev-images/epp:f1f87e9
```
same here

capri-xiyue committed:
change to install scheme

robscott commented on pkg/epp/controller/inferencepool_reconciler_test.go:
```diff
@@ -21,6 +21,9 @@ import (
 	"testing"
 	"time"
 
+	v1 "sigs.k8s.io/gateway-api-inference-extension/api/v1"
+	"sigs.k8s.io/gateway-api-inference-extension/apix/v1alpha2"
```
I think golangci-lint is going to complain about the placement of these imports - should be grouped with other imports from same package.

robscott commented on test/e2e/epp/e2e_suite_test.go:
```diff
@@ -77,10 +78,12 @@ const (
-	inferPoolManifest = "../../../config/crd/bases/inference.networking.x-k8s.io_inferencepools.yaml"
-	// inferModelManifest is the manifest for the inference model CRD.
-	inferModelManifest = "../../../config/crd/bases/inference.networking.x-k8s.io_inferencemodels.yaml"
+	// xInferPoolManifest is the manifest for the inference pool CRD with 'inference.networking.x-k8s.io' group.
+	xInferPoolManifest = "../../../config/crd/bases/inference.networking.x-k8s.io_inferencepools.yaml"
```
Do we still need this?

robscott reviewed (approved):
Thanks @capri-xiyue!

capri-xiyue committed:
fixed import

capri-xiyue committed:
change back to main image

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

kfswain commented:
Looks good to me! I dont think we have all the fun automation in your fork so feel free to merge whenev

capri-xiyue committed:
Merge branch 'capri-xiyue/v1-inference-pool' into capri-xiyue/v1-epp

AndresGuedez committed:
Add an "Implementing a Compatible Data Plane" section to the implementers guide (#1143)

* add an "Implementing a Compatible Data Plane" section to the implementers guide.

* minor cleanup

* update bullet list formatting.

* minor cleanup.

LukeAVanDrie committed:
feat(flowcontrol): Implement registry shard (#1187)

This commit introduces the core operational layer of the Flow Registry's
sharded architecture. It deliberately focuses on implementing the
`registryShard`—the concurrent, high-performance data plane for a single
worker—as opposed to the top-level `FlowRegistry` administrative
control plane, which will be built upon this foundation.

The `registryShard` provides the concrete implementation of the
`contracts.RegistryShard` port, giving each `FlowController` worker a
safe, partitioned view of the system's state. This design is
fundamental to achieving scalability by minimizing cross-worker
contention on the hot path.

The key components are:

- **`registry.Config`**: The master configuration blueprint for the
  entire `FlowRegistry`. It is validated once and then partitioned,
  with each shard receiving its own slice of the configuration,
  notably for capacity limits.

- **`registry.registryShard`**: The operational heart of this commit. It
  manages the lifecycle of queues and policies within a single shard,
  providing the read-oriented access needed by a `FlowController`
  worker. It ensures concurrency safety through a combination of mutexes
  for structural changes and lock-free atomics for statistics.

- **`registry.managedQueue`**: A stateful decorator that wraps a raw
  `framework.SafeQueue`. Its two primary responsibilities are to enable
  the sharded model by providing atomic, upwardly-reconciled statistics,
  and to enforce lifecycle state (active vs. draining), which is
  essential for the graceful draining of flows during future
  administrative updates.

- **Contracts and Errors**: New sentinel errors are added to the
  `contracts` package to create a clear, stable API boundary between
  the registry and its consumers.

This work establishes the robust, scalable, and concurrent foundation
upon which the top-level `FlowRegistry` administrative interface will be
built.

LukeAVanDrie committed:
feat(flowcontrol): refine types and docs (#1191)

This commit improves the foundational `types` package by consolidating
documentation and adding a safer default enum value in preparation for
the new `FlowController` implementation. It also removes concepts
related to displacement, which are out of scope for the initial
release.

Key changes:
- The package-level `README.md` has been removed. Its content is now
  in a comprehensive GoDoc package comment in `doc.go`. This addresses
  reviewer feedback to co-locate documentation with the code it
  describes, reducing maintenance burden and preventing doc drift.

- The package documentation has been rewritten to tell a clearer
  architectural narrative, explaining the request lifecycle through
  the lens of the `EnqueueAndWait` model.

- The `ErrDisplaced` error and `QueueOutcomeEvictedDisplaced` outcome
  have been removed to align the types with the GA feature set.

- A new `QueueOutcomeNotYetFinalized` enum value has been added. This
  serves as a safer, explicit zero-value for the `QueueOutcome` type,
  which is used by the new `ShardProcessor` to represent the initial
  state of a request before its lifecycle is complete.

nekomeowww committed:
docs: update to use kebab-cased flags changed at #1177 (#1193)

nirrozenbaum committed:
added graceful shutdown when scheduler config is not initialized (#1198)

Signed-off-by: Nir Rozenbaum <nirro@il.ibm.com>

capri-xiyue committed:
feat: move x-k8s to apix and add v1 InferencePool to api/v1 (#1116)

* rename x-k8s to apix and add v1 InferencePool to api/v1

* updated docker file

* fixed naming collision

* added fake approved annotation

* try to fix unit tests

* fixed typo

* added annotation

* fixed annotation

* Update api/v1/inferencepool_types.go

Co-authored-by: Rob Scott <rob.scott87@gmail.com>

* Apply suggestions from code review

Co-authored-by: Rob Scott <rob.scott87@gmail.com>

* use v1 type in v1alpha2

* clean unused depdendency

* fixed code generator error

* revert back

* merge bob's change

* upadted code-gen

* remove deep copy needs

* use codegen

* revert to workable version

* re-run generate

* revert to workable version

* change to use v1alpha2 back

* resolve merge conflicts

* fixed typo

* fixed typo

* fixed pipeline

* fixed boilerplate

Signed-off-by: Xiyue Yu <xiyue@google.com>

* updated missing dependency

* fixed change

* fixed format

* fixed import issue

* fixed format

---------

Signed-off-by: Xiyue Yu <xiyue@google.com>
Co-authored-by: Rob Scott <rob.scott87@gmail.com>

capri-xiyue committed:
resolve merge conflicts
